tedge/src/system_services/error.rs

```diff
@@ -0,0 +1,38 @@
+#[derive(thiserror::Error, Debug)]
+pub enum ServicesError {
+    #[error(transparent)]
+    IoError(#[from] std::io::Error),
+
+    #[error(transparent)]
+    SystemdError(#[from] SystemdError),
```

Comment:
It's a bit surprising to find systemd specific errors among the generic errors.

It seems to me that the errors under `SystemdError ` can be made independent of a given `SystemServiceManager`.

Reply:
Well, we are using the `#[error(...)]` attributes for `SystemdError` to present the user with error messages, and if we have generic errors, this would turn into a generic error message.